### ISSUE TITLE
Windows: Zig build linking error on gnu target

### ISSUE THREAD

**Issue by kassane** [ziglang/zig#8430]
Ref.: [lld-link error - _emutls](https://github.com/msys2/MINGW-packages/issues/8272)

**Comment by kassane** [comment]
Would it make it possible to choose to allow dynamic linking to lld?

**Comment by LemonBoy** [comment]
There are two problems here:
* Our emulated TLS implementation doesn't work on Windows
* You're targeting mingw and not cygwin (I guess?), LLVM should not use the emulated TLS

Do you have some more info and a minimal test case to reproduce the problem?

**Comment by kassane** [comment]
The environment the problem refers to is only msys2/mingw.
I tried in the first approach to use a build following the default configuration and the error shown in pkg1 occurs.
Later I readjusted it similar to the archlinux case and chose to link libclang dynamically, but now only lld is not being linked as shown in pkg2.

So in short, Zig does not accept to link with lld and libcompiler-rt available in the msys2 environment showing reference ambiguity during the linking process of the compiler itself.
When testing the working application, it turns out that msvc mode and among other targets work, but when directing windows-gnu there are linking problems.

See [PKGBUILD](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-zig/PKGBUILD)
